EventFilter/CastorRawToDigi/src/ZdcUnpacker.cc

```diff
@@ -0,0 +1,520 @@
+                  ncurr++;
+                  npre++;
+                  }
+                  }*/
+
+    //////////////////////////////////////////////
+    qie_begin=(HcalQIESample*)daq_first;
+    qie_end=(HcalQIESample*)(daq_last+1); // one beyond last..
+    // std::cout<<qie_end<<std::endl;
+    /// work through the samples
+    //currFiberChan=0x3F; // invalid fiber+channel...
+    //ncurr=0;
+    //valid=false;
+    //int myiter=0;
+/*	int spigot0 = 0;
+	int spigot1 = 0;
+	int spigotn1 = 0;
+	int spigothold=0;
+	
+	int htr0 = 0;
+	int htr1 = 0;
+	int htrhold=0;
+	
+	int fiberchannel[18];
+	for(int x = 0; x < 18; x++){
+	  fiberchannel[x]=0;
+	}
+	int fiberchannelextra = 0;
+	
+	int fiber[18];
+	for(int x = 0; x < 18; x++){
+	  fiber[x]=0;
+	}
+	int fiberextra = 0;
+*/	
+//    for (qie_work=qie_begin; qie_work!=qie_end;qie_work++) {
+    for (qie_work=qie_begin; qie_work!=qie_end;) {    
+      if (qie_work->raw()==0xFFFF) {
+        qie_work++;
+        continue; // filler word
+      }
+      // always at the beginning ...
+//      currFiberChan=qie_work->fiberAndChan();
+
+      // lookup the right channel
+      //HcalElectronicsId eid(qie_work->fiberChan(),qie_work->fiber(),spigot,22);
+//      CastorElectronicsId eid(qie_work->fiberChan(),qie_work->fiber(),spigot,3);
+   CastorElectronicsId eid(qie_work->fiberChan(),qie_work->fiber(),spigot,dccid);
+      eid.setHTR(htr_cr,htr_slot,htr_tb);
+  
+ /* if(spigot0%50000==0){
+    std::cout<<"spigot0 is "<<spigot0<<" and spigot1 is "<<spigot1<<" and spigotn1 is "<<spigotn1<<" and random is "<<spigothold<<std::endl;
+    std::cout<<"htr0 is "<<htr0<<" and htr1 is "<<htr1<<" and random is "<<htrhold<<std::endl;    
+    for(int x = 0; x < 18; x++){
+      std::cout<<"fiberchannel"<<x<<" is "<<fiberchannel[x]<<std::endl;
+    }
+    std::cout<<"fiberchannelextra is "<<fiberchannelextra<<std::endl;
+    for(int x = 0; x < 18; x++){
+      std::cout<<"fiber"<<x<<" is "<<fiber[x]<<std::endl;
+    }
+    std::cout<<"fiberextra is "<<fiberextra<<std::endl;
+    } 
+
+	if(spigot==0) spigot0++;
+	else if(spigot==1) spigot1++;
+	else if(spigot==-1) spigotn1++;
+	else spigothold++;
+
+	if(htr_tb==0) htr0++;
+	else if(htr_tb==1) htr1++;	
+	else htrhold++;
+	
+	for(int x = 0; x < 18; x++){
+	  if(qie_work->fiber()==x) fiber[x]++;
+	  else fiberextra++;
+	}
+	
+	for(int x = 0; x < 18; x++){
+	  if(qie_work->fiberChan()==x) fiberchannel[x]++;
+	  else fiberchannelextra++;
+	}
+	*/
+      //  std::cout<<qie_work->fiberChan()<<std::endl;
+      eid.setHTR(htr_cr,htr_slot,htr_tb);
+      //CastorElectronics
+      //      std::cout<<eid<<"with fiber channel .. "<<qie_work->fiberChan()<<" and fiber ... "<<qie_work->fiber()<<std::endl;
+
+//      std::cout<<"The fiberChan is ... "<<qie_work->fiberChan()<<" the fiber is ... "<<qie_work->fiber()<<" the spigot is ... "<<spigot<<" the htr_tb is... "<<htr_tb<<std::endl;
+
+
+      if (qie_work->fiberChan()==0 && qie_work->fiber()==1 && spigot==0 && htr_tb==1)
```

Comment:
what's the reason for this unfolded code?
It looks like the ```colls.zdcCont->push_back..; qie_work=ZdcUnpacker_impl::unpack... ;``` is all the same.
Just check the valid set of arguments and make just one call.